src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -508,6 +522,49 @@ private bool ConfigureLocale()
             return true;
         }
 
+        private bool SyncOptionalWorkloads()
+        {
+            bool isHiveUpdated = false;
+            bool isCustomHive = _commandInput.HasDebuggingFlag("--debug:ephemeral-hive") || _commandInput.HasDebuggingFlag("--debug:custom-hive");
+
+            if (!isCustomHive)
+            {
+
+                string sdkVersion = EnvironmentSettings.Host.Version;
+
+                try
+                {
+                    List<InstallationRequest> oWInstallationRequest = new List<InstallationRequest>();
+                    HashSet<string> syncRequestsPackageIds = new HashSet<string>();
+                    TemplateLocator optionalWorkloadLocator = new TemplateLocator();
+
+                    IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> syncRequests = optionalWorkloadLocator.GetDotnetSdkTemplatePackages(sdkVersion);
+                    foreach (IOptionalSdkTemplatePackageInfo packageInfo in syncRequests)
+                    {
+                        oWInstallationRequest.Add(new InstallationRequest(packageInfo.Path, true));
```

Comment:
nit: Please consider naming the bool argument for readability:
```suggestion
                        oWInstallationRequest.Add(new InstallationRequest(packageInfo.Path, isPartOfAnOptionalWorkload: true));
```

Reply:
fixed